### PR TITLE
GIPL Permafrost updates

### PR DESCRIPTION
**Pull request by BobTorgerson**
Updates GIPL Permafrost formatting for the summarized variables over a time range and creates an endpoint for gathering all of the time ranges desired by NCR. 

This PR will split out the summarized GIPL permafrost data out into each models min, mean, and maximum in the returned JSON.

You can see this in: http://localhost:5000/permafrost/point/gipl/65/-156/2021/2039/mmm

Additionally, this adds a new /ncr endpoint that combines all of the required data for the NCR Permafrost section from a single call to the API.

This can be seen here: http://localhost:5000/ncr/permafrost/point/65/-156

### CODE REVIEW

**Comment by BobTorgerson** [comment]
Closes #261

**Comment by charparr** [comment]
Now should close  Close #258

**Comment by charparr** [comment]
We may need to take a look at the CSV output, the columns will probably have to be:
model - summary - variable - value

image

**Review by charparr — COMMENTED** [review]
Bob this looks good! Thanks for cleaning this up. I took this opportunity to spruce up the documentation, and to fix the CSV output when the mmm summary is used to reflect that we are now not summarizing across models. I like how you've added the NCR route here as well, that is definitely convenient. This looks good from my perspective. I'll let you merge this, or we can keep it open for a bit if you like. The reason to maybe keep it open is because we'll be adding the IEM Historical here once that coverage is ready.

File: templates/permafrost/point.html
Comment (charparr) [inline templates/permafrost/point.html:18]: Thanks for adding this note Bob, I can see where this would be confusing!

File: routes/permafrost.py
Comment (charparr) [inline routes/permafrost.py:151]: I'm wondering if one of these dict keys should be "CRU-TS 4.0" to indicate the "model" so-to-speak used for the historical baseline?

**Review by charparr — APPROVED** [review]
Hey Bob this looks really good!

I tried quite a few different queries, for both CSV and JSON returns. Everything is working as I'd expect. I also tested the previous endpoint for the 4 km Melvin GIPL + other permafrost layers, and that service is intact. I think the documentation is looking for this endpoint as well.

File: routes/permafrost.py
Comment (BobTorgerson) [reply routes/permafrost.py]: Yes, you are correct that we should specify the model as CRU-TS 4.0 and the scenario as Historical. Thanks for catching this! I've added that change to the most recent commit.

**Review by brucecrevensten — COMMENTED** [review]
Points for discussion before merge,

 - can we reduce the number of round-trips to Rasdaman by a bunch (perhaps through removing min/max at this time),
 - chat with team about versioning for the API endpoints plus/and structure of API requests (like, api path structure) for the different flavors of permafrost we have.  I get that "old vs. new" are distinct right now but/and maybe we want to reflect a bit -- there's no right answer and it'd be valid to move ahead with how things are structured now
 - minor quibble with NCR endpoint, consider the main keys being the date ranges instead of numerical indices (look at temp/precip endpoints as example)

**Comment by cstephen** [comment]
This PR has been superseded by PR #266, which was based off of the `gipl_ncr_request` branch with some of its git history dropped, so I'm going to close this PR to avoid confusion.

**Pull request closed without merging**
